lti_consumer/lti_1p3/extensions/rest_framework/authentication.py

```diff
@@ -0,0 +1,79 @@
+"""
+Django REST Framework extensions for LTI 1.3 & LTI Advantage implementation.
+
+Implements a custom authentication class to be used by LTI Advantage extensions.
+"""
+from django.utils.translation import ugettext as _
+from rest_framework import authentication
+from rest_framework import exceptions
+
+from lti_consumer.models import LtiConfiguration
+
+
+class Lti1p3ApiAuthentication(authentication.BaseAuthentication):
+    """
+    LTI 1.3 Token based authentication.
+
+    Clients should authenticate by passing the token key in the "Authorization".
+    LTI 1.3 expects a token like the following:
+        Authorization: Bearer jwt-token
+
+    Since the base implementation of this library uses JWT tokens, we expect
+    a RSA256 signed token that contains the allowed scopes.
+    """
+    keyword = 'Bearer'
+
+    def authenticate(self, request):
+        """
+        Authenticate an LTI 1.3 Tool.
+
+        This doesn't return a user, but let's the external access and commit
+        changes.
+        TODO: Consider creating an user for LTI operations, both to keep track
+        of changes and to use Django's authorization flow.
+        """
+        auth = request.headers.get('Authorization', '').split()
+        lti_config_id = request.parser_context['kwargs'].get('lti_config_id')
+
+        # Check if auth token is present on request and is correctly formatted.
+        if not auth or auth[0].lower() != self.keyword.lower():
+            msg = _('Missing LTI 1.3 authentication token.')
+            raise exceptions.AuthenticationFailed(msg)
+
+        if len(auth) == 1:
+            msg = _('Invalid token header. No credentials provided.')
+            raise exceptions.AuthenticationFailed(msg)
+
+        if len(auth) > 2:
+            msg = _('Invalid token header. Token string should not contain spaces.')
+            raise exceptions.AuthenticationFailed(msg)
+
+        # Retrieve LTI configuration or fail if it doesn't exist
+        try:
+            lti_configuration = LtiConfiguration.objects.get(pk=lti_config_id)
+            lti_consumer = lti_configuration.get_lti_consumer()
+        except:
```

Comment:
nit. This is still missing the `Exception`

Reply:
Thanks for catching this one!